rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceConfigurationParser.java

```diff
@@ -0,0 +1,140 @@
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client.nodesource;
+
+import java.util.HashMap;
+
+import org.ow2.proactive_grid_cloud_portal.rm.client.NodeSourceConfiguration;
+import org.ow2.proactive_grid_cloud_portal.rm.client.PluginDescriptor;
+import org.ow2.proactive_grid_cloud_portal.rm.client.RMController;
+
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONException;
+import com.google.gwt.json.client.JSONObject;
+
+
+public class NodeSourceConfigurationParser {
+
+    private RMController controller;
+
+    public NodeSourceConfigurationParser(RMController controller) {
+        this.controller = controller;
+    }
+
+    public NodeSourceConfiguration parseNodeSourceConfiguration(String json) throws ImportException {
+
+        JSONObject jsonObject;
+
+        try {
+            jsonObject = this.controller.parseJSON(json).isObject();
+        } catch (JSONException e) {
+            throw new ImportException("The imported node source is not a valid JSON file", e);
+        }
+
+        try {
+            String nodeSourceName = jsonObject.get("nodeSourceName").isString().stringValue();
+            boolean nodesRecoverable = jsonObject.get("nodesRecoverable").isBoolean().booleanValue();
+
+            JSONObject infrastructurePluginDescriptorJson = jsonObject.get("infrastructurePluginDescriptor").isObject();
+            String infrastructurePluginName = infrastructurePluginDescriptorJson.get("pluginName")
+                                                                                .isString()
+                                                                                .stringValue();
+            PluginDescriptor infrastructurePluginDescriptor = getPluginDescriptor(infrastructurePluginDescriptorJson,
+                                                                                  infrastructurePluginName);
+
+            JSONObject policyPluginDescriptorJson = jsonObject.get("policyPluginDescriptor").isObject();
+            String policyPluginName = policyPluginDescriptorJson.get("pluginName").isString().stringValue();
+            PluginDescriptor policyPluginDescriptor = getPluginDescriptor(policyPluginDescriptorJson, policyPluginName);
+
+            return new NodeSourceConfiguration(nodeSourceName,
+                                               nodesRecoverable,
+                                               infrastructurePluginDescriptor,
+                                               policyPluginDescriptor);
+        } catch (RuntimeException e) {
+            throw new ImportException("The imported node source has incorrect parameters.", e);
+        }
+    }
+
+    public HashMap<String, PluginDescriptor> parsePluginDescriptors(String json) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)